pystac/extensions/eo.py

```diff
@@ -365,8 +365,12 @@ def bands(self) -> Optional[List[Band]]:
         """Get or sets a list of :class:`~pystac.Band` objects that represent
         the available bands.
         """
+
+        def construct_bands(bands: List[Dict[str, Any]]) -> List[Band]:
+            return [Band(b) for b in bands]
+
         return map_opt(
-            lambda bands: [Band(b) for b in bands],
+            construct_bands,
```

Comment:
It looks like the reason that mypy can infer a type for the lambda in the `_get_bands` method above but not here is due to the way the 2nd argument to `map_opt` is typed.

In this case, we didn't have a type annotation on `SummariesExtension` for the `summaries` property. Adding the following type annotation in `pystac.extensions.base.SummariesExtension` allows us to keep an untyped lambda here:

```python
from pystac.collection import Collection, Summaries

...

class SummariesExtension:
    summaries: Summaries
```

Since it seems like `mypy` is actually catching cases where we haven't adequately typed things, this might be a better solution that creating a typed function.